sky/global_user_state.py

```diff
@@ -408,3 +548,38 @@ def get_storage() -> List[Dict[str, Any]]:
             'status': StorageStatus[status],
         })
     return records
+
+
+def get_cost_for_cluster(cluster_name: str,) -> float:
+
+    handle = get_handle_from_cluster_name(cluster_name)
+    cluster_hash = get_cluster_hash_from_name(cluster_name)
+    metadata = get_cluster_history_metadata(cluster_hash)
+
+    usage_intervals = metadata['usage_intervals']
+
+    return get_cost_for_usage_intervals(handle, usage_intervals)
+
+
+def get_cost_for_usage_intervals(
+    handle: Optional['backends.Backend.ResourceHandle'],
```

Comment:
Instead of using the `handle` to get the resources and price, we can use the `pickle.loads(launched_resources)` from the `cluster_history` table we saved above. In this way, the function can be more future-proof -- able to generate the cost report for the terminated clusters.